cloudconfig/cloudinit/network_ubuntu.go

```diff
@@ -183,6 +186,17 @@ func GenerateNetplan(interfaces corenetwork.InterfaceInfos) (string, error) {
 		if info.MTU != 0 && info.MTU != 1500 {
 			iface.MTU = info.MTU
 		}
+
+		// There is a disjoint between KVM and LXD behaviour here.
+		// In KVMs the kernel assigns a device name different to the one we
+		// configured, so we need to match on the hardware address.
+		// For LXD, it has been observed that 22.04 containers sometimes get a
+		// different *MAC* than the one we configured, so we don't want to
+		// match by MAC, rather the default, which is the device ID.
```

Comment:
'rather the default' is a bit awkward.

containers sometimes get a different *MAC* than the one we configured, so we match by name rather than MAC.

I feel like we really need to dig if the MAC isn't matching, because that feels like a real bug we should be squashing rather than working around.